### PR TITLE
Handle writing of empty arrays to preCICE

### PR DESCRIPTION
**Pull request by IshaanDesai**
In a parallel coupling case, ranks which have no part of the coupling interface will call the write data functionality with empty data structures. The bindings need to handle this as the main preCICE API allows this.

### CODE REVIEW

**Comment by BenjaminRueth** [comment]
I converted this PR to a draft, since there are still some errors in the source code that the tests run into (see Travis). Do you already want me to do a review now or do you first fix the errors?

**Review by BenjaminRueth — CHANGES_REQUESTED** [review]
There are some formatting changes in `setup.py`. If they are necessary due to code style, please directly commit them to develop. Don't add them to this PR.

**Comment by IshaanDesai** [comment]
> There are some formatting changes in `setup.py`. If they are necessary due to code style, please directly commit them to develop. Don't add them to this PR.

Formatting changes to `setup.py` were introduced unintentionally during a merge conflict resolution, sorry about that. The reformatting is now done in develop and then `develop` is merged into this branch.

**Comment by IshaanDesai** [comment]
> I converted this PR to a draft, since there are still some errors in the source code that the tests run into (see Travis). Do you already want me to do a review now or do you first fix the errors?

The error in tests was just due to a small spelling mistake. It is now rectified and PR is ready for review :+1:

**Review by BenjaminRueth — CHANGES_REQUESTED** [review]
Looks good. Minor point regarding the non-`block` operations. I also added some tests in 86fc5d4 that worked nicely.

I also checked that the new tests actually fail for the current state of `develop` and got the following error:
```
======================================================================
ERROR: test_read_write_block_scalar_data_empty (test_bindings_module.TestBindings)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test/test_bindings_module.pyx", line 201, in test_bindings_module.TestBindings.test_read_write_block_scalar_data_empty
    solver_interface.write_block_scalar_data(1, [], write_data)
  File "precice.pyx", line 952, in precice.Interface.write_block_scalar_data
    size = vertex_ids.size
AttributeError: 'list' object has no attribute 'size'

======================================================================
ERROR: test_read_write_block_vector_data_empty (test_bindings_module.TestBindings)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test/test_bindings_module.pyx", line 237, in test_bindings_module.TestBindings.test_read_write_block_vector_data_empty
    solver_interface.write_block_vector_data(1, [], write_data)
  File "precice.pyx", line 871, in precice.Interface.write_block_vector_data
    size, dimensions = values.shape
ValueError: need more than 1 value to unpack

----------------------------------------------------------------------
```

File: precice.pyx [hunk 1]
```diff
@@ -913,8 +917,10 @@ cdef class Interface:
         """
         if not isinstance(value, np.ndarray):
             value = np.asarray(value)
-        dimensions = value.size
-        assert(dimensions == self.get_dimensions())
+        if len(value) > 0:
+            dimensions = value.size
+            assert(dimensions == self.get_dimensions())
+
```
[the remark this answers is not in the document]
Comment (uekerman) [reply precice.pyx:923]: Yes, I agree with @BenjaminRueth's

**Pull request merged**